LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -237,7 +241,31 @@ public void CreateClassrooms(int classroomCount)
 
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+
+            string action = "Assign Instructors To Classrooms";
+            ReportHeader(actionName: action);
+
+            if (classroomList.Count == 0)
+            {
+                ReportError(missingResource: "Instructor", failedAction: action);
+            }
+            else
+            {
+                foreach (var classroom in classroomList)
+                {
+                    if (ModuleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                    {
+                        List<Instructor> instructorList;
+                        ModuleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
```

Comment:
You have to check the instructorList size, otherwise if instructorList.Count== 0 
-- you will get an exception on calling
instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];